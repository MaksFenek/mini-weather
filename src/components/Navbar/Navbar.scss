@import url('https://fonts.googleapis.com/css2?family=Patrick+Hand&display=swap');
.nav {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  background: #e3f2fd;
  height: 60px;
  font-size: 2rem;
  font-family: 'Patrick Hand', cursive;
  padding: 0 calc(50% - 590px);
  padding-left: 14px;

  &-logo {
    display: flex;
    align-items: center;

    img,
    h1 {
      cursor: pointer;
    }

    h1 {
      margin-left: 24px;
      line-height: 0;
    }
  }

  &-search {
    margin-left: 24px;
  }
}
