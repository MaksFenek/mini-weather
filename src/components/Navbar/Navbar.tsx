import React from 'react';
import './Navbar.scss';
import Input from '@material-ui/core/Input';

interface INavbar {
  onKeyPress: (e: React.KeyboardEvent<HTMLInputElement>) => void;
  searchRef: any;
}

const Navbar: React.FC<INavbar> = ({ onKeyPress, searchRef }) => {
  return (
    <nav className='nav' aria-label='nav'>
      <div className='nav-logo'>
        <img src='favicon.png' alt='Logo' height='48px' width='48px' />
        <h1>Weather</h1>
      </div>
      <div className='nav-search'>
        <Input
          inputRef={searchRef}
          onKeyPress={onKeyPress}
          placeholder='Search...'
          inputProps={{ 'aria-label': 'search', id: 'autocomplete' }}
          type='text'
        />
      </div>
    </nav>
  );
};

export default Navbar;
