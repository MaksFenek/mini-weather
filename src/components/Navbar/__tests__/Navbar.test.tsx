import React from 'react';
import Navbar from '../Navbar';
import renderer from 'react-test-renderer';

describe('<Navbar />', () => {
  const onKeyPress = jest.fn();
  const ref = null;

  const tree = renderer
    .create(<Navbar onKeyPress={onKeyPress} searchRef={ref} />)
    .toJSON();

  it('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
