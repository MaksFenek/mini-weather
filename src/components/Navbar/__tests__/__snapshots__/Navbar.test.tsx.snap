// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render correctly 1`] = `
<nav
  aria-label="nav"
  className="nav"
>
  <div
    className="nav-logo"
  >
    <img
      alt="Logo"
      height="48px"
      src="favicon.png"
      width="48px"
    />
    <h1>
      Weather
    </h1>
  </div>
  <div
    className="nav-search"
  >
    <div
      aria-label="search"
      className="MuiInputBase-root MuiInput-root MuiInput-underline"
      onClick={[Function]}
      onKeyPress={[MockFunction]}
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <input
        className="MuiInputBase-input MuiInput-input"
        id="autocomplete"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Search..."
        type="text"
      />
    </div>
  </div>
</nav>
`;
